helsingborg-stad/app-mitt-helsingborg
# feat: adjust signin to be able to fit notifications later on

## Explain the changes you’ve made

Changed the authenticating logic in the application. From start the authentication flow where initiated from splash screen, which is not changed from being initialised from the AuthContext instead. In the context we check if the user is signed in or not and setting the new authState (former isUserAuthenticated state). The new authState has 3 states, PENDING, SIGNED_IN and SIGNED_OUT, which describes more correct in which state the users auth state is in. 

By having these 3 new states, we can easier decide which screens to render, i.e. splash, signed out or signed in screens.

Also did change som files to have correct file extensions 

Removed unused OnboardingScreen component

Removed unused dependency in package.json

## Explain why these changes are made
To easier know in which state the auth state of a user is in, and also this changes make it simpler to implement notifications, which should be done later on.

## How to test the changes?

Concrete example:
1. Checkout this branch
2. Make sure that the USE_BANKID in .env file is set to true
3. Make sure you have a test BankID user and bankid-api up and running
4. Test login, logout in the application

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
